### PR TITLE
Remove hook leech

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Restriction of #6268 restricted to leech part for simplicity

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java [hunk 1]
```diff
@@ -38,8 +40,8 @@
 import com.ichi2.libanki.Decks;
 import com.ichi2.libanki.Note;
 import com.ichi2.libanki.Utils;
-import com.ichi2.libanki.hooks.Hooks;
 
+import com.ichi2.libanki.hooks.Leech;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:44]: Codacy Issue found: [Avoid unused imports such as 'com.ichi2.libanki.hooks.Leech'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575774)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575774&bid=18280061)

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java [hunk 2]
```diff
@@ -22,13 +22,15 @@
 import android.app.Activity;
 
 import android.content.Context;
+import android.content.res.Resources;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:25]: Codacy Issue found: [Avoid unused imports such as 'android.content.res.Resources'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575774)

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java [hunk 3]
```diff
@@ -22,13 +22,15 @@
...
 import android.database.Cursor;
 import android.database.SQLException;
 import android.database.sqlite.SQLiteConstraintException;
 import android.graphics.Typeface;
 import android.text.SpannableStringBuilder;
 import android.text.TextUtils;
 import android.text.style.StyleSpan;
+import android.widget.Toast;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:33]: Codacy Issue found: [Avoid unused imports such as 'android.widget.Toast'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575774)

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java [hunk 4]
```diff
@@ -36,8 +36,8 @@
 import com.ichi2.libanki.Decks;
 import com.ichi2.libanki.Note;
 import com.ichi2.libanki.Utils;
-import com.ichi2.libanki.hooks.Hooks;
 
+import com.ichi2.libanki.hooks.Leech;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/Sched.java:40]: Codacy Issue found: [Avoid unused imports such as 'com.ichi2.libanki.hooks.Leech'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575774)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575774&bid=18280061)

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java [hunk 5]
```diff
@@ -22,6 +22,7 @@
 import android.app.Activity;
 
 import android.content.Context;
+import android.content.res.Resources;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java:25]: Codacy Issue found: [Avoid unused imports such as 'android.content.res.Resources'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575774)

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java [hunk 6]
```diff
@@ -234,4 +239,25 @@ public String toString() {
...
+            }
+            activity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    Toast.makeText(activity, leechMessage, Toast.LENGTH_SHORT).show();
+                }
+            });
+
+        } else {
+            Timber.e("LeechHook :: could not show leech toast as activity was null");
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java:260]: Ideally leave in a separate commit as it's a functionality change - has been an issue for some time.

```suggestion
            Timber.w("LeechHook :: could not show leech toast as activity was null");
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java [hunk 7]
```diff
@@ -234,4 +239,25 @@ public String toString() {
     public interface CountMethod {
         int operation(long did, int lim);
     }
+
+    protected static void leech(Card card, Activity activity) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java:243]: Do you think this would make more sense as a static in the Leech class? You know the scope of `Sched` better than anyone here.

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java:243]: Dealing with leech is really an action only done while scheduling. I believe that this is the place where it makes the most sens.
The only reason I can see for a hook in leech, upstream, is that showing the tooltip is a front-end action and not a back-end one. So the code is in the reviewer and not in the scheduler.
I could also put this in the Reviewer as a static method, and call it from Scheduler. I really don't see the point of a whole class just to show a toast

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java (hunk 6 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java:260]: I'm okay with adding this change if it please you. I've no idea what commit message to write. I do not know the reason of this change. Can you write the message please ?

**Comment by david-allison-1** [comment]
@Arthur-Milchior I can't reply to your comment.

LOG or NF prefix:

`NF: Leech - Reduced Log Level`

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java (hunk 6 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java:260]: LOG or NF prefix:

`NF: Leech - Reduced Log Level`
